### ISSUE TITLE
for a ensemble model how to construct the space?

### ISSUE THREAD

**Issue by ROOKLO** [hyperopt/hyperopt#781]
Hi, I want to tune the Adaboost classifier's parameters, an ensemble model with a default decision tree classifier. If I want to tune the parameter for both of them simultaneously, how should I construct the space? Or tune them another way.

`dt = DecisionTreeClassifier(max_depth=None,
                            min_samples_split=2,
                            min_samples_leaf=1)

adaboost_clf = AdaBoostClassifier(base_estimator=dt,
                              n_estimators=50,
                              learning_rate=1.0,
                              algorithm='SAMME.R')`

**Comment by ROOKLO** [comment]
`ada_space = {
    # 'base_estimator':DecisionTreeClassifier(),
    "n_estimators": hp.choice("ada_n_estimators", [100, 200, 300, 400, 500, 600]),
    'learning_rate':hp.quniform('ada_learning_rate', 0.1, 1., 0.01),
    'algorithm':hp.choice('ada_algorithm', ['SAMME.R','SAMME'])
}


dt_space = {
    'max_depth':hp.quniform('dt_max_depth', [1, 7, 1]),
    'min_samples_split':hp.quniform('dt_min_samples_split', [2, 10, 1]),
    'min_samples_leaf':hp.quniform('dt_min_samples_leaf', [1, 10, 1])

}

spaces = dict({'ada':ada_space, 'dt':dt_space})

def hyperparameter_tuning(params):
    DT = DecisionTreeClassifier(**params['dt'])
    clf = AdaBoostClassifier(base_estimator=DT ,**params['ada'])
    acc = cross_val_score(clf, X_scaled, y_train,scoring="accuracy").mean()
    return {"loss": -acc, "status": STATUS_OK}`

For me, it works!

**Comment by ROOKLO** [comment]
`ada_space = {
    # 'base_estimator':DecisionTreeClassifier(),
    "n_estimators": hp.choice("ada_n_estimators", [100, 200, 300, 400, 500, 600]),
    'learning_rate':hp.quniform('ada_learning_rate', 0.1, 1.0, 0.05),
    'algorithm':hp.choice('ada_algorithm', ['SAMME.R','SAMME'])
}

dt_space = {
    'max_depth':hp.quniform('dt_max_depth', 1, 7, 1),
    'min_samples_split':hp.quniform('dt_min_samples_split', 0.01, 1.0, 0.01),
    'min_samples_leaf':hp.quniform('dt_min_samples_leaf', 0.01, 0.5, 0.01)
}

spaces = dict({'ada':ada_space, 'dt':dt_space})

def hyperparameter_tuning(params):
    DT = DecisionTreeClassifier(**params['dt'])
    clf = AdaBoostClassifier(base_estimator=DT ,**params['ada'])
    acc = cross_val_score(clf, X_scaled, y_train,scoring="accuracy").mean()
    return {"loss": -acc, "status": STATUS_OK}`

For me, it works!!
